rero_ils/modules/notifications/api.py

```diff
@@ -187,12 +188,12 @@ def replace_pids_and_refs(self):
             patron = Patron.get_record_by_pid(data['loan']['patron']['pid'])
             view_code = patron.get_organisation().get('code')
             base_url = current_app.config.get('RERO_ILS_APP_URL')
-            profile_url = f'{base_url}/{view_code}/patrons/profile'
+            next_page_url = urllib.parse.quote(f'{view_code}/patrons/profile')
+            profile_url = f'{base_url}/signin/?next={next_page_url}'
             data['loan']['profile_url'] = profile_url
```

Comment:
I don't think this is the right profile url. The url is correct as it was before, but we have to get redirected to the signin if we are not logged in.

Reply:
I think we have to change this:
```
def login_and_patron():
    """Patron is logged in."""
    if current_user and not current_user.is_authenticated:
        abort(401)
    if len(current_patrons) == 0:
        abort(403)
```